app/services/compute_resource_host_importer.rb

```diff
@@ -0,0 +1,70 @@
+class ComputeResourceHostImporter
+  attr_accessor :host, :compute_resource, :vm
+
+  delegate :logger, :to => :Rails
+
+  def initialize(opts = {})
+    self.compute_resource = opts.fetch(:compute_resource)
+    self.vm = compute_resource.find_vm_by_uuid(opts[:uuid]) if opts[:uuid]
+    self.vm ||= opts.fetch(:vm)
+    self.host = Host::Managed.new(
+      :managed => true,
+      :build => false,
+      :compute_resource_id => compute_resource.id,
+      :vm => vm
```

Comment:
Today I was trying to import a virtual machine, for a Host that already existed. I expected the import code would associate the host and the VM, however it attempted to create a new one (and failed because they had the same fqdn). Can you check that? I think the only required thing would be to change this `new` call to

`.where(....properties....).first_or_initialize`